li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerImpl.java

```diff
@@ -271,6 +292,24 @@ private LiKafkaProducerImpl(LiKafkaProducerConfig configs,
         _auditor.record(auditToken, topic, timestamp, 1L, 0L, AuditType.ATTEMPT);
         throw t;
       }
+
+      if (partition == null && _partitioner != null) {
```

Comment:
what happens if partition == null and _partitioner == null?

Reply:
that means the client (the underlying one) is using the default partitioner, which we know is safe